tests/test_factory.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) QuantCo 2025-2025
+# SPDX-License-Identifier: BSD-3-Clause
+
+import polars as pl
+import pytest
+from polars.testing import assert_frame_equal
+
+import dataframely as dy
+from dataframely._filter import Filter
+from dataframely.testing import create_collection
+
+# ----------------------------------- Schemas And Collections ----------------------------------- #
+
+
+class MySchema(dy.Schema):
+    a = dy.Integer(primary_key=True)
+
+
+class MySchema2(dy.Schema):
+    a = dy.Integer(primary_key=True)
+    b = dy.String(nullable=True)
+
+
+class MyCollection(dy.Collection):
+    member: dy.LazyFrame[MySchema]
+
+    @dy.filter()
+    def a_smaller_ten(self) -> pl.LazyFrame:
+        return self.member.filter(pl.col("a") < 10)
+
+    def foo(self) -> str:
+        return "foo"
+
+
+# ------------------------------------------ Fixtures ------------------------------------------- #
+
+
+@pytest.fixture()
+def member() -> pl.LazyFrame:
+    return pl.LazyFrame({"a": [0, 1, 2, 3, 10]})
+
+
+@pytest.fixture()
+def member2() -> pl.LazyFrame:
+    return pl.LazyFrame({"a": [0, 1, 2, 3, 10], "b": ["a", "b", "c", "d", "e"]})
```

Comment:
I'm not a fan of data in fixtures as changing the fixture for a test requires updating all other tests. Can we just hardcode these in the tests themselves and duplicate them?